src/components/Course/CoursesList/index.tsx

```diff
@@ -18,13 +21,14 @@ import {
 } from '@/utils/navigation';
 
 type Props = {
-  courses: Course[];
+  initialResponse: CoursesResponse;
   isMyCourses: boolean;
 };
 
 const MIN_COURSES_COUNT = 6;
+const ROWS_BEFORE_END = 4;
 
-const CoursesList: React.FC<Props> = ({ courses, isMyCourses }) => {
+const CoursesList: React.FC<Props> = ({ initialResponse, isMyCourses }) => {
```

Comment:
There are 2 ways to do the infinite scrolling without a lot of manual logic:

1. Using `swr/infinite` e.g. `src/components/Collection/CollectionDetailContainer/CollectionDetailContainer.tsx`
2. Using `react-virtuoso` e.g. `src/components/Notes/NotesPage/Tabs/MyPrivateNotesTab/NotesList/index.tsx` or `src/components/Notes/NotesPage/Tabs/MyPublicReflectionsTab/UserPublicReflectionsList/index.tsx`

We should use either of them. I am leaning towards `swr/infinite`.